src/testFixtures/java/org/opensearch/neuralsearch/BaseNeuralSearchIT.java

```diff
@@ -812,6 +817,63 @@ private String registerModelGroup() {
         return modelGroupId;
     }
 
+    // Method that waits till the health of nodes in the cluster goes green
+    protected void waitForClusterHealthGreen(String numOfNodes) throws IOException {
+        Request waitForGreen = new Request("GET", "/_cluster/health");
+        waitForGreen.addParameter("wait_for_nodes", numOfNodes);
+        waitForGreen.addParameter("wait_for_status", "green");
+        client().performRequest(waitForGreen);
+    }
+
+    /**
+     * Add a single Doc to an index
+     *
+     * @param index name of the index
+     * @param docId
+     * @param fieldName name of the field
+     * @param text to be added
+     */
+    protected void addDocument(String index, String docId, String fieldName, String text) throws IOException {
+        Request request = new Request("PUT", "/" + index + "/_doc/" + docId + "?refresh=true");
+
+        XContentBuilder builder = XContentFactory.jsonBuilder().startObject().field(fieldName, text).endObject();
+        request.setJsonEntity(builder.toString());
+        client().performRequest(request);
+
+        request = new Request("POST", "/" + index + "/_refresh");
```

Comment:
Do we need this if we call refresh in the put request?

Reply:
No we don't need it